chore: clarify how to get internal NLB (#2356)
kubernetes-sigs/aws-load-balancer-controller: docs/guide/service/annotations.md

- <a name="lb-type">`service.beta.kubernetes.io/aws-load-balancer-type`</a> specifies the load balancer type. This controller reconciles those service resources with this annotation set to either `nlb-ip` or `external`.

    !!!tip

>         This annotation specifies the controller used to provision LoadBalancers (as specified in [legacy-cloud-provider](#legacy-cloud-provider)). Refer to [lb-scheme](#lb-scheme) to specify whether the LoadBalancer is public-facing or internal.

Reviewer: ```suggestion
        This annotation specifies the controller used to provision LoadBalancers (as specified in [legacy-cloud-provider](#legacy-cloud-provider)). Refer to [lb-scheme](#lb-scheme) to specify whether the LoadBalancer is Internet-facing or internal.
```

>         This annotation specifies the controller used to provision LoadBalancers (as specified in [legacy-cloud-provider](#legacy-cloud-provider)). Refer to [lb-scheme](#lb-scheme) to specify whether the LoadBalancer is Internet-facing or internal.

Reviewer 2: ```suggestion
        This annotation specifies the controller used to provision LoadBalancers (as specified in [legacy-cloud-provider](#legacy-cloud-provider)). Refer to [lb-scheme](#lb-scheme) to specify whether the LoadBalancer is Internet-facing or internal.
```
```suggestion
        This annotation specifies the controller used to provision LoadBalancers (as specified in [legacy-cloud-provider](#legacy-cloud-provider)). Refer to [lb-scheme](#lb-scheme) to specify whether the LoadBalancer is internet-facing or internal.
```

    
    !!!note ""

>         - [Deprecated] `nlb-ip` type, controller will provision NLB with IP targets. This value is supported for backwards compatibility.
>         - For `external` type, NLB target type depend on the annotation [nlb-target-type](#nlb-target-type).

Reviewer: ```suggestion
        - [Deprecated] For type `nlb-ip`, the controller will provision an NLB with targets registered by IP address. This value is supported for backwards compatibility.
        - For type `external` type, the NLB target type depends on the [nlb-target-type](#nlb-target-type) annotation.
```

Author: Duplicated "type" in "For type `external` type"

Reviewer: Ouch. Thank you for catching that.

- <a name="lb-type">`service.beta.kubernetes.io/aws-load-balancer-type`</a> specifies the load balancer type. This controller reconciles those service resources with this annotation set to either `nlb-ip` or `external`.

    !!!note ""
        - For `nlb-ip` type, controller will provision NLB with IP targets. This value is supported for backwards compatibility, but it is deprecated.

>         - For `external` type, NLB target type depend on the annotation [nlb-target-type](#nlb-target-type). If used in combination with [lb-scheme](#lb-scheme) set to internal, it will provision an internal NLB.

Reviewer 3: Thanks for contributing this.
I think maybe ```- [Deprecated] `nlb-ip` type, controller will provision NLB with IP targets. This value is supported for backwards compatibility```

BTW, it feels a weird to mention  lb-scheme here as it have it's own section. and internal NLB is the default mode if lb-scheme is not specified after all.

Author: 1st thing :+1: 

About the lb-scheme, I just wanted to highlight that "external" allows to use "internal" LBs. That's what happened to me, I did not try to use the external type because I thought that would only provision an internet-facing LB :sweat_smile:

Maybe if not a note in the lb-scheme clarifying that it can be used with the "external" type?

Reviewer 3: @froblesmartin 
I understand your confusing now. it's a valid point. I think maybe add a separate tip section like 
```
!!!tip
   This annotation specifies the controller used to provision LoadBalancers. Refer to [lb-scheme](#lb-scheme) to specify whether the LoadBalancer is public-facing.
```

Reviewer 3: btw, this annotation itself will be deprecated soon as well, we'll use new spec.ServiceClass in the future instead of `service.beta.kubernetes.io/aws-load-balancer-type`

Author: Good to know! :smile: I have changed it to the tip, and as well reduced the last part just to clarify that this controller can provision any NLB+Target Group configuration.

The AWS Load Balancer Controller manages Kubernetes Services in a compatible way with the legacy aws cloud provider. The annotation `service.beta.kubernetes.io/aws-load-balancer-type` is used to determine which controller reconciles the service. If the annotation value is `nlb-ip` or `external`, legacy cloud provider ignores the service resource (provided it has the correct patch) so that the AWS Load Balancer controller can take over. For all other values of the annotation, the legacy cloud provider will handle the service. Note that this annotation should be specified during service creation and not edited later.

The legacy cloud provider patch was added in Kubernetes v1.20 and is backported to Kubernetes v1.18.18+, v1.19.10+.

!!!note ""

>     With the right combination of annotations any Network Load Balancer and Target Group configuration can be provisioned with this controller.

Reviewer 2: This note doesn't provide additional information. Lets remove it.

Author: > The AWS Load Balancer Controller manages Kubernetes Services in a compatible way with the legacy aws cloud provider. The annotation `service.beta.kubernetes.io/aws-load-balancer-type` is used to determine which controller reconciles the service. If the annotation value is `nlb-ip` or `external`, legacy cloud provider ignores the service resource (provided it has the correct patch) so that the AWS Load Balancer controller can take over. For all other values of the annotation, the legacy cloud provider will handle the service. Note that this annotation should be specified during service creation and not edited later.

As with the main issue, right after reading that paragraph, I assumed that the new controller can not manage an internal load balancer with instance target type. If you are not okay adding that note, maybe I could rewrite that paragraph to make that clearer?

Reviewer 2: sounds better

> 
> !!!note ""
>     With the right combination of annotations any Network Load Balancer and Target Group configuration can be provisioned with this controller.

Reviewer 2: ```suggestion

!!!note ""
    With the right combination of annotations any Network Load Balancer and Target Group configuration can be provisioned with this controller.
```
```suggestion

```
